### PR TITLE
Created base background Component.

### PR DESCRIPTION
**Pull request by tk4chenko**
The BaseBackgroundView class is a subclass of UIView that provides some additional design features that can be customized in Interface Builder. This class is marked with the @IBDesignable attribute, which allows it to be rendered directly in Interface Builder.

The class includes the following @IBInspectable properties that can be customized in the Attribute Inspector of Interface Builder:
* borderWidth: Sets the width of the border of the view.
* cornerRadius: Sets the corner radius of the view.
* distance: Sets the distance between the two borders of the view.
* borderColor: Sets the color of the border of the view.

The addDoubleBorder method is responsible for adding the second border to the view. It creates a new UIView instance called bottomView, sets its layer properties and background color, removes needless lines of bottomView using a mask and add it as a subview on parent view. The new border view is positioned slightly below and to the right of the main view, creating the desired double border effect.

To use the CustomView class in project, follow these steps:
* In Interface Builder, drag a new UIView onto your storyboard or xib file.
* Select the new view, and in the Identity Inspector, set its class to BaseBackgroundView.
* In the Attributes Inspector, you will see the four new properties that you can customize: borderWidth, cornerRadius, distance, and borderColor. Adjust these properties to achieve the desired visual effect.
Screenshot 2023-02-26 at 22 56 12
Screenshot 2023-02-26 at 22 56 26

### CODE REVIEW

**Comment by Dorogobid** [comment]
We see needless lines when a transparent background is configured.

Screenshot 2023-02-24 at 22 51 36

**Comment by tk4chenko** [comment]
> We see needless lines when a transparent background is configured.
> 
> Screenshot 2023-02-24 at 22 51 36



> We see needless lines when a transparent background is configured.
> 
> Screenshot 2023-02-24 at 22 51 36

**Comment by tk4chenko** [comment]
> We see needless lines when a transparent background is configured.
> 
> Screenshot 2023-02-24 at 22 51 36

What is the problem to set the background color?

**Comment by tk4chenko** [comment]
Removed needless lines of bottomView using a mask. Now you can see the bottomView in Interface Builder, but when you load the project in the simulator, you will only see the part that is the border .

File: Skylab/Custom elements/CustomView.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
+//
+//  CustomView.swift
+//  Skylab
+//
+//  Created by Artem Tkachenko on 22.02.2023.
+//
+
+import UIKit
+
+@IBDesignable class CustomView: UIView {
```
Comment (Dorogobid) [inline Skylab/Custom elements/CustomView.swift:10]: You need to think about the name of the component class. There are can be a few CustomViews in our project.

**Review by Dorogobid — APPROVED** [review]
Cool realization using BezierPath

File: Skylab/Custom elements/CustomView.swift (hunk 1 above)
Comment (tk4chenko) [reply Skylab/Custom elements/CustomView.swift:10]: I think we will all decide which name is best.

Comment (alexslkgit) [reply Skylab/Custom elements/CustomView.swift:10]: I would propose BaseBackgroundView or MainBackgroundView or something like this :) 

But I fully agree with Serhiy, you to add some names during development, and if someone doesn't like your idea, he can leave a comment on your PR.

File: Skylab/Custom elements/CustomView.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (alexslkgit) [inline Skylab/Custom elements/CustomView.swift:11]: BTW according to the US, an extension should be created, but now I think what you've done is probably a better idea  👍

**Review by alexslkgit — CHANGES_REQUESTED** [review]

**Review by artembilyy — APPROVED** [review]
Good

**Review by alexslkgit — APPROVED** [review]

**Review by SerhiiMolodets — APPROVED** [review]

File: Skylab/Custom elements/BaseBackgroundView.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+//
+//  Created by Artem Tkachenko on 22.02.2023.
+//
+
+import UIKit
+
+@IBDesignable class BaseBackgroundView: UIView {
+    
+    @IBInspectable var borderWidth: CGFloat = 0 {
+        didSet { self.layer.borderWidth = borderWidth }
```
[the remark this answers is not in the document]
Comment (tk4chenko) [reply Skylab/Custom elements/BaseBackgroundView.swift:14]: Yes, I did that.

**Pull request merged**